internal/exec/resolvable/resolvable.go

```diff
@@ -253,66 +289,95 @@ func (b *execBuilder) makeObjectExec(typeName string, fields schema.FieldList, p
 
-func (b *execBuilder) makeFieldExec(typeName string, f *schema.Field, m reflect.Method, methodIndex int, methodHasReceiver bool) (*Field, error) {
-	in := make([]reflect.Type, m.Type.NumIn())
-	for i := range in {
-		in[i] = m.Type.In(i)
-	}
-	if methodHasReceiver {
-		in = in[1:] // first parameter is receiver
-	}
-
-	hasContext := len(in) > 0 && in[0] == contextType
-	if hasContext {
-		in = in[1:]
-	}
+func (b *execBuilder) makeFieldExec(typeName string, f *schema.Field, m reflect.Method, sf reflect.StructField,
+	methodIndex, fieldIndex int, methodHasReceiver bool) (*Field, error) {
 
 	var argsPacker *packer.StructPacker
-	if len(f.Args) > 0 {
-		if len(in) == 0 {
-			return nil, fmt.Errorf("must have parameter for field arguments")
+	var hasError bool
+	var hasContext bool
+
+	// Validate resolver method only when there is one
+	if methodIndex != -1 {
+		in := make([]reflect.Type, m.Type.NumIn())
+		for i := range in {
+			in[i] = m.Type.In(i)
 		}
-		var err error
-		argsPacker, err = b.packerBuilder.MakeStructPacker(f.Args, in[0])
-		if err != nil {
-			return nil, err
+		if methodHasReceiver {
+			in = in[1:] // first parameter is receiver
 		}
-		in = in[1:]
-	}
 
-	if len(in) > 0 {
-		return nil, fmt.Errorf("too many parameters")
-	}
+		hasContext = len(in) > 0 && in[0] == contextType
+		if hasContext {
+			in = in[1:]
+		}
 
-	if m.Type.NumOut() > 2 {
-		return nil, fmt.Errorf("too many return values")
-	}
+		if len(f.Args) > 0 {
+			if len(in) == 0 {
+				return nil, fmt.Errorf("must have parameter for field arguments")
+			}
+			var err error
+			argsPacker, err = b.packerBuilder.MakeStructPacker(f.Args, in[0])
+			if err != nil {
+				return nil, err
+			}
+			in = in[1:]
+		}
+
+		if len(in) > 0 {
+			return nil, fmt.Errorf("too many parameters")
+		}
 
-	hasError := m.Type.NumOut() == 2
-	if hasError {
-		if m.Type.Out(1) != errorType {
-			return nil, fmt.Errorf(`must have "error" as its second return value`)
+		if m.Type.NumOut() > 2 {
+			return nil, fmt.Errorf("too many return values")
+		}
+
+		hasError = m.Type.NumOut() == 2
+		if hasError {
+			if m.Type.Out(1) != errorType {
+				return nil, fmt.Errorf(`must have "error" as its second return value`)
+			}
 		}
 	}
 
 	fe := &Field{
 		Field:       *f,
 		TypeName:    typeName,
 		MethodIndex: methodIndex,
+		FieldIndex:  fieldIndex,
 		HasContext:  hasContext,
 		ArgsPacker:  argsPacker,
 		HasError:    hasError,
 		TraceLabel:  fmt.Sprintf("GraphQL field: %s.%s", typeName, f.Name),
 	}
-	if err := b.assignExec(&fe.ValueExec, f.Type, m.Type.Out(0)); err != nil {
+
+	var out reflect.Type
+	if methodIndex != -1 {
+		out = m.Type.Out(0)
+	} else {
+		out = sf.Type
+	}
+	if err := b.assignExec(&fe.ValueExec, f.Type, out); err != nil {
 		return nil, err
 	}
+
 	return fe, nil
 }
 
+// find method with the same name or one which has `Resolver` suffix
 func findMethod(t reflect.Type, name string) int {
+	resName := name + "Resolver"
```

Comment:
This is completely unrelated change (i.e. find method with name `<field>Resolver`). Please, send this in a separate PR. I'd personally, vote 👎 for this specific change, since adds unnecessary complexity to the library. Less is more in this case.

Reply:
This is not an unrelated change. As mentioned in the PR notes, when a struct implements an interface, then struct's resolver methods are used for each interface field. Hence, when a struct has a field `Email` and interface has the same field, then there will be a conflict between struct field and resolver method. So a `Resolver` suffix can be used to resolve that conflict. If you can suggest something else to achieve the same goal then I will be happy to implement that

Reply:
As a workaround you can just use different struct fields. If you are implementing an interface, it's just the methods that will be used, right? This would mean that you can use differnt field names.